feat: Export Results as Dataframe

priyadarshini-ni opened the pull request:
- [ ] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nisystemlink-clients-python/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

This PR adds utility to export Results as pandas Dataframe.
1. The user can pass filter and column projection for querying the results.

### Why should this Pull Request be merged?

This utility can be useful to get the results data in the form of a Dataframe which can be useful for further processing.

### What testing has been done?

1. Automated tests have been added

Madhan-Reddy-ni committed:
First Pass

Madhan-Reddy-ni committed:
Added the examples

Madhan-Reddy-ni committed:
Removed the unnecessary changes

adamarnesen committed:
feat: Add client for SystemLink products API (#69)

Commit:
1.8.0

Automatically generated by python-semantic-release

Madhan-Reddy-ni committed:
Fix: PR Comments

Madhan-Reddy-ni committed:
Fix: Linting

mure committed:
chore: remove @mure as code owner

rbell517 committed:
chore: add @rbell517 as code owner (#92)

AMoldova-NI committed:
fix: Update to the latest httpx version (#88)

Commit:
1.8.1

Automatically generated by python-semantic-release

rbell517 committed:
chore: Add python 3.13 to tested versions (#93)

Giriharan219 committed:
feat: Add minimal client implementation for Feeds API (#73)

Commit:
1.9.0

Automatically generated by python-semantic-release

santhoshramaraj committed:
feat: example to upload file to SystemLink (#81)

Commit:
1.10.0

Automatically generated by python-semantic-release

Commit:
Merge branch 'master' into users/Peram/niresults/add-results-client

Giriharan219 committed:
test: Stabilize FeedsClient integration tests (#95)

Commit:
fix: resolve PR comments

Commit:
fix: rename file, add model

Commit:
add retry for 503, 504

Commit:
add workspace update parameter

Commit:
fix: lint

Commit:
remove arg for update-results

Commit:
remove is_finalized

Commit:
fix: lint

Commit:
fix: add updateResultRequest

Commit:
move the results client into testmonitor client

Commit:
update doc

Commit:
add mapping for model in test

Commit:
fix: resolve comments

Commit:
fix: add model for create request and partial response

Commit:
move result example to testmonitor folder

Commit:
fix: nit changes, added standard status types

Commit:
fix: type error

Commit:
fix: move status to a separate file

Commit:
add files and package

Commit:
add normlize method

Commit:
add init file

Commit:
add tests

Commit:
file restructure

Commit:
fix: test cases

Commit:
fix: lint

SSadaiyappan-NI commented on nisystemlink/clients/testmonitor/utilities/_client_utilities.py:
```diff
@@ -0,0 +1,45 @@
+)
+
+
+def __batch_query_results(
```
exposing this method as well could be useful. @SSSantosh18  your thoughts? we could probably do the same for product and spec as well

SSadaiyappan-NI commented on nisystemlink/clients/testmonitor/utilities/_dataframe_utilities.py:
```diff
@@ -0,0 +1,54 @@
+    query_filter: str,
+    column_projection: Optional[List[ResultProjection]] = None,
+) -> pd.DataFrame:
+    """Fetches results of a specific product and normalizes them into a Pandas DataFrame.
```
```suggestion
    """Fetches test results and normalizes them into a Pandas DataFrame.
```

priyadarshini-ni replied:
Updated

SSadaiyappan-NI commented on nisystemlink/clients/testmonitor/utilities/_client_utilities.py:
```diff
@@ -0,0 +1,45 @@
+    """Fetches results of a specific product in batches.
+
+    Args:
+        client: The TestMonitorClient to fetch results data.
```
```suggestion
        client: The TestMonitorClient to use for the request.
```

SSadaiyappan-NI commented on nisystemlink/clients/testmonitor/utilities/_client_utilities.py:
```diff
@@ -0,0 +1,45 @@
+    query_filter: str,
+    column_projection: Optional[List[ResultProjection]] = None,
+) -> List[Result]:
+    """Fetches results of a specific product in batches.
```
```suggestion
    """Fetches test results in batches.
```

priyadarshini-ni replied:
Updated

SSadaiyappan-NI commented on nisystemlink/clients/testmonitor/utilities/_client_utilities.py:
```diff
@@ -0,0 +1,45 @@
+            Fields you do not specify are excluded. Returns all fields if no value is specified.
+
+    Returns:
+        List[Results]: A list of results.
```
Lets not mention the type, we should stay consistent

priyadarshini-ni replied:
Removed type

SSadaiyappan-NI commented on nisystemlink/clients/testmonitor/utilities/_dataframe_utilities.py:
```diff
@@ -0,0 +1,54 @@
+
+    Args:
+        client: The TestMonitorClient to fetch results data.
+        query_filter: The filter to use when querying the results.
```
```suggestion
        query_filter: The result query filter in Dynamic Linq format.
```

priyadarshini-ni replied:
Updated

SSadaiyappan-NI commented on nisystemlink/clients/testmonitor/utilities/_dataframe_utilities.py:
```diff
@@ -0,0 +1,54 @@
+    """Fetches results of a specific product and normalizes them into a Pandas DataFrame.
+
+    Args:
+        client: The TestMonitorClient to fetch results data.
```
```suggestion
        client: The TestMonitorClient to use for the request.
```

priyadarshini-ni replied:
Updated

SSadaiyappan-NI commented on nisystemlink/clients/testmonitor/utilities/_dataframe_utilities.py:
```diff
@@ -0,0 +1,54 @@
+        A Pandas DataFrame containing the results.
+
+    Raises:
+        ApiException: If unable to communicate with the ``/nitestmonitor`` service
```
```suggestion
        ApiException: If unable to communicate with the `/nitestmonitor` service
```

priyadarshini-ni replied:
Updated

SSadaiyappan-NI commented on tests/integration/testmonitor/test_testmonitor.py:
```diff
@@ -391,8 +394,100 @@ def test__update_properties_append__properties_appended(
+            query_response.results
+        )
+
+        results_dataframe = get_results_dataframe(
+            client, query_filter=query_results_filter
+        )
+
```
```suggestion
```

priyadarshini-ni replied:
Removed

SSadaiyappan-NI commented on tests/integration/testmonitor/test_testmonitor.py:
leaving blank lines only between sections of arrange, act and assert. Don't leave blank lines in the same block.

priyadarshini-ni replied:
Removed unnecessary blank lines

Commit:
resolve pr comments

Commit:
fix: lint

Shri2109 committed:
feat: Add and resolve query projection for products and specs respectively (#97)

Co-authored-by: Sam rishi <sam.rishi@solitontech.local>
Co-authored-by: sam-rishi <155733081+sam-rishi@users.noreply.github.com>
Co-authored-by: Sam Rishi <sam.rishi@ni.com>

BREAKING CHANGE: Product and Specification models have been updated
to support projections.
 - Product client changes--
   - `models.Product` now defines all fields as Optional.
   - `ProductClient.create_product`'s `products` parameter is now typed
     as `models.CreateProductRequest`
   - `ProductClient.update_product`'s `products` parameter is now typed
     as `models.UpdateProductRequest`
 - Specifications client changes--
   - `models.Specification` and `models.SpecificationDefinition` now
     define all fields as Optional.
   - `models.QuerySpecifications` has been renamed `models.PagedSpecifications`
     to better align to other clients.
   - `models.CreateSpecificationeRequest.specs` is now typed as `models.CreateSpecificationsRequestObject`
     instead of `models.SpecificationDefinition`
   - `models.UpdateSpecificationeRequest.specs` is now typed as `models.UpdateSpecificationsRequestObject`
     instead of `models.Specification`

Commit:
2.0.0

Automatically generated by python-semantic-release

Commit:
move constant to class

Madhan-Reddy-ni committed:
feat: Add client for SystemLink results API (#82)

Co-authored-by: Priyadarshini Piramanayagam <priydarshini.piramanayagam@emerson.com>

Commit:
Merge remote-tracking branch 'upstream/master' into users/priya/feat/dataframe-utility-for-results
